src/libcore/cell.rs

```diff
@@ -1022,12 +1023,15 @@ impl<'b> BorrowRef<'b> {
     #[inline]
     fn new(borrow: &'b Cell<BorrowFlag>) -> Option<BorrowRef<'b>> {
         let b = borrow.get();
-        if b >= MIN_WRITING {
+        if b < UNUSED {
             None
         } else {
-            // Prevent the borrow counter from overflowing into
-            // a writing borrow.
-            assert!(b < MIN_WRITING - 1);
+            // NOTE: In debug mode, overflow here will panic. In release mode,
+            // this might overflow into a writing borrow, but that's fine. A
+            // writing borrow will simply prevent more borrows of any kind from
+            // being created. The only way to get a writing borrow when one
+            // already exists is to use RefMut::map_split, but we know that
+            // there cannot be any RefMuts in existence right now.
```

Comment:
But what if you call this `2^31` *more* times (on a 32bit platform) to get the counter back to 0?

Oh, that's not possible because it will hit the `b < UNUSED` above. Maybe the comment should be extended to explain that.

Reply:
Fixed.